[Alerting V2][Serverless & 9.5][M2] Add experimental alerting features rules pages (#6523)
elastic/docs-content: explore-analyze/alerting/kibana-alerting-experimental/rules.md

---
navigation_title: Rules
applies_to:
  stack: experimental 9.5+
  serverless: experimental
products:
  - id: kibana
description: "Rules in Kibana's experimental alerting system define what to detect using ES|QL. Evaluation runs on a schedule; alerts, action policies, and notifications flow from rule detections."
---

# Rules in the {{alerting-v2-system}}

A rule is where the {{alerting-v2-system}} starts. It points {{kib}} at the data you care about, describes what counts as a problem in {{esql}}, and says how often to check. Alerts, action policies, and notifications all flow from what a rule detects. 

This page explains what rules do, what they don't control, and how to choose a creation path.

## What rules do [detection-and-notification]

> On each run, a rule executes an {{esql}} query against your data. If the query finds a match and the rule is in Signal mode, it writes a _signal_, a point-in-time record that the condition was met. In Alert mode, it also maintains an _alert episode_ for each matched series, tracking state from first breach through recovery.

Author: Applying [open comment](https://github.com/elastic/docs-content/pull/5528/changes#r3229529885) from Tia:

```suggestion
On each run, a rule executes an {{esql}} query against your data. If the query finds a match, it writes a rule event (`rule_event`), a point-in-time record that the condition was met. In Alert mode, it also maintains an _alert episode_ for each matched series, tracking state from first breach through recovery.
```
